src/components/GlobalPreferences/CustomLabels/ShareModal.js

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useRef, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Button,
+  ButtonIcon,
+  GU,
+  IconCopy,
+  Modal,
+  TextInput,
+  Toast,
+  breakpoint,
+  useTheme,
+} from '@aragon/ui'
+
+const TIMEOUT_TOAST = 4000
+
+function ShareModal({ inputRef, onClose, onCopy, onFocus, link, visible }) {
+  const theme = useTheme()
+
+  return (
+    <Modal visible={visible} onClose={onClose}>
+      <header
+        css={`
+          font-size: 22px;
+          line-height: 38px;
+          font-weight: bold;
+        `}
+      >
+        Share custom labels
+      </header>
+      <main style={{ marginTop: `${2 * GU}px` }}>
+        <div
+          css={`
+            font-size: 15px;
+            line-height: 22px;
+          `}
+        >
+          These labels will be shared with everyone that has access to this
+          link.
+        </div>
+        <div style={{ marginTop: `${2.5 * GU}px` }}>
+          <div
+            css={`
+              font-size: 12px;
+              line-height: 16px;
+              text-transform: uppercase;
+              color: ##6d777b;
+            `}
+          >
+            Link
+          </div>
+          <div
+            css={`
+              display: inline-flex;
+              max-width: 100%;
+              width: 100%;
+              position: relative;
+              background: ${theme.surface};
+              border: 1px solid ${theme.border};
+              border-radius: 3px;
+              box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.06);
+              padding-right: 30px;
+              margin-top: ${1 * GU}px;
+            `}
+          >
+            <TextInput
+              ref={inputRef}
+              value={link}
+              onFocus={onFocus}
+              readOnly
+              css={`
+                text-overflow: ellipsis;
+                width: 100%;
+                max-width: 100%;
+                border: 0;
+                box-shadow: none;
+                background: ${theme.surface};
+                &:read-only {
+                  color: ${theme.content};
+                  text-shadow: none;
+                }
+              `}
+            />
+            <ButtonIcon
+              onClick={onCopy}
+              label="Copy to clipboard"
+              css={`
+                position: absolute;
+                top: 0;
+                right: 0;
+                width: 39px;
+                height: 38px;
+                display: flex;
+                align-items: center;
+                justify-content: center;
+                border-radius: 0 3px 3px 0;
+                &:active {
+                  background: rgba(220, 234, 239, 0.3);
```

Comment:
Does this take care of the alpha value too?

Reply:
I don’t think there is an alpha value on these? Otherwise, if you need alpha values, you can now do `.alpha(0.3)` on any color coming from the theme :+1:

Reply:
🍬 🍬